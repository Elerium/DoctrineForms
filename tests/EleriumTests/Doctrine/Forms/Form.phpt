<?php

/**
 * Part of Elerium Framework
 * Copyright (c) 2013
 */
 
namespace EleriumTests\Doctrine\Forms;

use Elerium;
use EleriumTests\Mocks;
use Tester\Assert;
use Elerium\Doctrine\Forms;

require_once __DIR__ . '/../../../bootstrap.php';
require_once __DIR__ . '/../../../Mocks/EntityManager.php';
require_once __DIR__ . '/entities.php';

$entityManager = new Mocks\EntityManager;

Assert::throws(function() use ($entityManager) {
	new Forms\Form($entityManager, NULL);
}, 'Elerium\InvalidArgumentException', "Entity class must be string, NULL given.");

Assert::throws(function() use ($entityManager) {
	new Forms\Form($entityManager, 1);
}, 'Elerium\InvalidArgumentException', "Entity class must be string, integer given.");

Assert::throws(function() use ($entityManager) {
	new Forms\Form($entityManager, 'EleriumTests\Doctrine\Forms\Entity\InvalidEntity');
}, 'Elerium\InvalidArgumentException', "Entity class 'EleriumTests\\Doctrine\\Forms\\Entity\\InvalidEntity' does not exist.");

new Forms\Form($entityManager, 'EleriumTests\Doctrine\Forms\Entity\Person');
new Forms\Form($entityManager, new Entity\Person);