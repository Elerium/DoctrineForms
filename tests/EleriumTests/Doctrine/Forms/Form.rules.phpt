<?php

/**
 * Part of Elerium Framework
 * Copyright (c) 2013
 */
 
namespace EleriumTests\Doctrine\Forms;

use Elerium;
use EleriumTests\Mocks;
use Tester\Assert;
use Elerium\Doctrine\Forms,
	Doctrine\DBAL\Types\Type,
	Elerium\Doctrine\Forms\Types\Email,
	Doctrine\Common\Annotations\SimpleAnnotationReader;

require_once __DIR__ . '/../../../bootstrap.php';
require_once __DIR__ . '/../../../Mocks/EntityManager.php';
require_once __DIR__ . '/entities.php';

$reader = new SimpleAnnotationReader;
require_once ELERIUM_DIR . '/Doctrine/Forms/Mapping/Form.php';
$reader->addNamespace('Elerium\Doctrine\Forms\Mapping'); // Require to be loaded

Type::addType(Email::NAME, 'Elerium\Doctrine\Forms\Types\Email');

$form = new Forms\Form(new Mocks\EntityManager, new Entity\PersonDetail, $reader);
$form->addText('id')->addRule(Forms\Form::UNIQUE, 'ID already exist');
$form->addText('name');
$form->addText('age');
$form->addText('height');
$form->addText('email');

Assert::same(file_get_contents(__DIR__ . '/Form.rules.expected'), $form->__toString());