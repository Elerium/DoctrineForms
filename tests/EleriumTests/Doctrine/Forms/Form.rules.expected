<form action="" method="post">

<table>
<tr class="required">
	<th><label class="required" for="frm-id"></label></th>

	<td><input type="text" class="text" name="id" id="frm-id" required="required" data-nette-rules='{op:&#39;:integer&#39;,msg:&#39;Wrong type!&#39;},{op:&#39;:filled&#39;,msg:&#39;Please complete mandatory field.&#39;},{op:"Elerium\\Doctrine\\Forms\\Rules::validateUnique",msg:&#39;ID already exist&#39;}' value="" /></td>
</tr>

<tr class="required">
	<th><label class="required" for="frm-name"></label></th>

	<td><input type="text" maxlength="64" class="text" name="name" id="frm-name" required="required" data-nette-rules="{op:':maxLength',msg:'Max length must be 64!',arg:64},{op:':filled',msg:'Please complete mandatory field.'}" value="" /></td>
</tr>

<tr>
	<th><label for="frm-age"></label></th>

	<td><input type="text" class="text" name="age" id="frm-age" data-nette-rules="{op:':integer',msg:'Please enter a numeric value.'}" value="" /></td>
</tr>

<tr>
	<th><label for="frm-height"></label></th>

	<td><input type="text" class="text" name="height" id="frm-height" data-nette-rules="{op:':float',msg:'Please enter a numeric value.'}" value="" /></td>
</tr>

<tr>
	<th><label for="frm-email"></label></th>

	<td><input type="text" class="text" name="email" id="frm-email" data-nette-rules="{op:':email',msg:'Please enter a valid email address.'}" value="" /></td>
</tr>
</table>

</form>
